### ISSUE TITLE
F1 + F2 without ARFCN

### ISSUE THREAD

**Issue by marvinmarnold** [marvinmarnold/meteor-imsi-catcher#2]
How can this be done?

**Comment by He3556** [comment]
yes you are right, we can not fingerprint like this with the Android API. 
Basically F1 and F2 tries to identify a "new" BTS, that was not there before. 
So (without arfcn) we had: 
F1 - same CID and LAC (like the original BTS)
F2 - other CID and LAC  (than every BTS around) 

To differentiate between the general and the Android view, i would add another column 
 "Android API case" - right next to the column "Details" - you think that is ok? 

The arfcn could have helped us to notice a new BTS - now we need another indicator for a new cell (BTS)
What about the time as a indicator? 
Let's say x (24) hours exploration time for an area where the user resists (not moving around for x (60) minutes). After the 24 hrs. a new CellID would rise an alarm. 
AIMSICD works with the OpenCellID database for detecting "unknown" Cells. I would suggest to work with an local db that the phone is collecting before we think about using an external db.

**Comment by marvinmarnold** [comment]
> To differentiate between the general and the Android view, i would add another column
> "Android API case" - right next to the column "Details" - you think that is ok?

Do you mean in the documentation? Sure. 
The current implementation also creates different types of readings. For example, a SIMReading is different than a GSMReading. It already looks like we'll have to disagregate GSMReading into AndroidV1Reading, AndroidV18GSMReading, AndriodV18LTEReading, etc. 

> What about the time as a indicator? 

That can work, but it sounds complex. Tracking whether or not somebody has moved and dealing with timers. I'd rather focus on some easier detections to start with then come back to this.

> AIMSICD works with the OpenCellID database for detecting "unknown" Cells. I would suggest to work with an local db that the phone is collecting before we think about using an external db. 

This is an important point. meteor-imsi-catcher is itself a DB of cell readings from multiple users. For StingWatch, I think you are right that it is simplest to forget about opencellid, and focus on readings that are collected through the app (by anybody). If we want to do data sharing w/ OCID down the line, that is still possible.

But does the fact that users are already sharing data change your proposal? What if a user enters somewhere that they've never been before but that somebody else has spent >24h in before? Does the answer change if the previous visit was 1 year ago or 1 day ago?

**Comment by He3556** [comment]
•••••••••••••••••••••••••••••••
Marcus Prem
0171.1775663
info@eventgroup-kl.de
<send_via_mobile>

> Am 22.02.2016 um 01:07 schrieb marvinmarnold <notifications@github.com>:
> 
> To differentiate between the general and the Android view, i would add another column
> "Android API case" - right next to the column "Details" - you think that is ok?
> 
> Do you mean in the documentation? Sure.  
> 
ok
> The current implementation also creates different types of readings. For example, a SIMReading is different than a GSMReading. It already looks like we'll have to disagregate GSMReading into AndroidV1Reading, AndroidV18GSMReading, AndriodV18LTEReading, etc.
> 
yes we have to do it this way 
> What about the time as a indicator?
> 
> That can work, but it sounds complex. Tracking whether or not somebody has moved and dealing with timers. I'd rather focus on some easier detections to start with then come back to this.
> 
agree
> AIMSICD works with the OpenCellID database for detecting "unknown" Cells. I would suggest to work with an local db that the phone is collecting before we think about using an external db.
> 
> This is an important point. meteor-imsi-catcher is itself a DB of cell readings from multiple users. For StingWatch, I think you are right that it is simplest to forget about opencellid, and focus on readings that are collected through the app (by anybody). If we want to do data sharing w/ OCID down the line, that is still possible.
> 
sharing is a nice idea here - but at least we need anonymization, encrytion and better also an authentication of the data. 
> But does the fact that users are already sharing data change your proposal? What if a user enters somewhere that they've never been before but that somebody else has spent >24h in before?
> 
as long as the data is valide it makes no difference who it was. But with local db i thought about a single phone.
> Does the answer change if the previous visit was 1 year ago or 1 day ago?
> 

It makes a difference how old the data is. It depends on how fast the network is growing or changing (updates) in the area. 
We could check openCellId db / history to get a feeling for it. 

But if the network is growing fast this detection would not work very good. 
If this is the case we had to deactivate the detection. 

But in my city i never noticed a new Cell or any change in the last 3 - 4  years.

**Comment by marvinmarnold** [comment]
> But with local db i thought about a single phone.

This is tricky with meteor/cordova. the options for local storage are 1) minimongo local only collection - not clear to me how long information is persisted here. 2) [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API) - this works well for things like preferences but doesn't scale well, 3) [web sql db](https://www.w3.org/TR/webdatabase/) - even though there is already a [cordova plugin](https://github.com/litehelpers/Cordova-sqlite-storage), it seems like this would be a lot of work. we would need to reimplement all mongo DB logic for this, 4) force offline data collection with something like [ground](https://atmospherejs.com/ground/db) and selectively sync.

I'd be most inclined to pursue #4, but that would be more to save battery than it would probably be to save battery, not just for the purposes of having a local collection. In any case, we can filter readings locally based on deviceID (below) and get the same effect.

> sharing is a nice idea here - but at least we need anonymization, encrytion and better also an authentication of the data.

anonymization - currently, a randomID is generated client side and sent with each reading. we could make it more anonymous by getting rid of the ID but then it will be harder to make sense of the readings.
encryption - in motion, everything will be done over HTTPS. at rest, information is currently in clear text. should open anohter issue if we want to address that.
authentication - what could be used for this? in order to keep things anonymous, there's not really a good mechanism to make sure someone isn't flooding w/ fake data.

**Comment by He3556** [comment]
ok, so it already does share the data? so it works like a cloud solution? 
So if we do all the detection in the cloud we don't really need a local db - i think No.1 would be the standard way to save some data transfer and speed up the presentation on the frontend?

**Comment by marvinmarnold** [comment]
Yes, it does sync data to server over DDP. I think most detections can be done on the client side only. the server 'cloud' would be optimized for detections that require analyzing readings from multiple sources.

**Comment by He3556** [comment]
sounds great - probably the server could even locate the catcher :)
